### PR TITLE
Change AI linked storage data source type to enum

### PR DESCRIPTION
**Pull request by olterner**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2020-03-01-preview/componentLinkedStorageAccounts_API.json [hunk 1]
```diff
@@ -282,7 +282,14 @@
       "in": "path",
       "required": true,
       "type": "string",
-      "description": "The type of the Application Insights component linked storage account.",
+      "enum": [
+        "ServiceProfiler"
+      ],
+      "x-ms-enum": {
+        "name": "StorageType",
+        "modelAsString": false
```
Comment (MyronFanQiu) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2020-03-01-preview/componentLinkedStorageAccounts_API.json]: usually we recommend setting it to true. Is there any context we need to know?

**Review by mmyyrroonn — APPROVED** [review]

**Review by pilor — APPROVED** [review]

**Comment by MyronFanQiu** [comment]
@akning-ms Hi~ Could you help us force merge this PR since there is a breaking change.

**Comment by olterner** [comment]
Can this be merged?

**Comment by MyronFanQiu** [comment]
@akning-ms This PR changes a string to an enum. The original PR was merged two weeks ago and the service is still under development. The server only accepts one enum string.

**Pull request merged**